internal/hardware/validator.go

```diff
@@ -147,25 +148,39 @@ func (v *validator) DiskIsEligible(ctx context.Context, disk *models.Disk, infra
 			fmt.Sprintf(wrongDriveTypeTemplate, disk.DriveType, strings.Join(v.getValidDeviceStorageTypes(hostArchitecture, clusterVersion), ", ")))
 	}
 
-	// We only allow multipath if all paths are FC
 	if disk.DriveType == models.DriveTypeMultipath {
-		for _, inventoryDisk := range inventory.Disks {
-			if lo.Contains(strings.Split(inventoryDisk.Holders, ","), disk.Name) {
-				if inventoryDisk.DriveType != models.DriveTypeFC {
-					notEligibleReasons = append(notEligibleReasons,
-						fmt.Sprintf(wrongMultipathTypeTemplate, inventoryDisk.DriveType, string(models.DriveTypeFC)))
-					break
+		fcDisks := hostutil.GetDisksOfHolderByType(inventory.Disks, disk, models.DriveTypeFC)
+		iSCSIDisks := hostutil.GetDisksOfHolderByType(inventory.Disks, disk, models.DriveTypeISCSI)
+		// We only allow multipath if all paths are FC/ iSCSI
+		if len(fcDisks) == 0 && len(iSCSIDisks) == 0 {
+			notEligibleReasons = append(notEligibleReasons,
+				fmt.Sprintf(wrongMultipathTypeTemplate, strings.Join([]string{string(models.DriveTypeFC), string(models.DriveTypeISCSI)}, ", ")))
+			// Return since further checks are unnecessary
+			return notEligibleReasons, nil
+		}
+		// We only allow multipath if all paths are of the same type
+		if len(fcDisks) > 0 && len(fcDisks) != len(hostutil.GetAllDisksOfHolder(inventory.Disks, disk)) {
+			notEligibleReasons = append(notEligibleReasons, mixedTypesInMultipath)
```

Comment:
```suggestion
			return append(notEligibleReasons, mixedTypesInMultipath), nil
```
I think we can return here and below as well?

Reply:
I considered that, but if there are issues with the iSCSI disks, wouldn’t it be better to display all the issues to the user at once?

Reply:
right :+1: , we can let it like that!